<?xml version="1.0"?>
<sdf version="1.4">
	<model name="M200">
		<pose>0 0 0 0 0 0</pose>
		<static>false</static>

		<link name="base_link">
			<inertial>
				<pose>0 0 0 0 0 0</pose>
				<mass>29.0</mass>
				<!-- Calculated using simple 3D rectangle model-->
				<inertia> 
					<ixx>2.417633</ixx>       
						<ixy>0.0</ixy>         
						<ixz>0.0</ixz>         
					<iyy>4.501041</iyy>       
						<iyz>0.0</iyz>         
					<izz>6.725341</izz>       
				</inertia>
			</inertial>

			<collision name="collision">
				<pose>"0 0 0 0 0 0</pose>
				<geometry>
					<box>
						<size>1.35 0.98 0.2</size>
					</box>
				</geometry>
			</collision>

			<visual name="visual">
				<pose>"0 0 0.09 1.57079 0 1.57079</pose>
				<geometry>
					<mesh>
						<uri>model://M200/meshes/kingfisher_base.stl</uri>
					</mesh>
				</geometry>
			</visual>

			<gravity>1</gravity>

		</link>

		<!-- IMU Sensor-->
		<link name="imu_link">
			<inertial>
				<pose>0 0 0 0 0 0</pose>
				<mass>1e-5</mass>
				<inertia> 
					<ixx>1e-6</ixx>       
						<ixy>0.0</ixy>         
						<ixz>0.0</ixz>         
					<iyy>1e-6</iyy>       
						<iyz>0.0</iyz>         
					<izz>1e-6</izz>       
				</inertia>
			</inertial>

			<collision name="collision">
				<pose>0 0 0 0 0 0</pose>
				<geometry>
					<box>
						<size>0.1 0.1 0.1</size>
					</box>
				</geometry>
			</collision>


			<sensor name="imu" type="imu">
				<pose>0 0 0 0 0 0</pose>
				<always_on>1</always_on>
				<update_rate>1000</update_rate>
			</sensor>

    	</link>
			
		<joint name="imu_joint" type="fixed">

			<parent>base_link</parent>
			<child>imu_link</child>
			<pose>0 0 0.079226 0 0 0</pose>	
		</joint>


		<!-- IMU Sensor-->
		<link name="gps_link">
			<inertial>
				<pose>0 0 0 0 0 0</pose>
				<mass>1e-5</mass>
				<inertia> 
					<ixx>1e-6</ixx>       
						<ixy>0.0</ixy>         
						<ixz>0.0</ixz>         
					<iyy>1e-6</iyy>       
						<iyz>0.0</iyz>         
					<izz>1e-6</izz>       
				</inertia>
			</inertial>

			<collision name="collision">
				<pose>0 0 0 0 0 0</pose>
				<geometry>
					<box>
						<size>0.1 0.1 0.1</size>
					</box>
				</geometry>
			</collision>


			<sensor name="gps_sensor" type="gps">
				<pose>0 0 0 0 0 0</pose>
				<always_on>1</always_on>
				<update_rate>5</update_rate>
			</sensor>

    	</link>

		<joint name="gps_joint" type="fixed">
			<parent>base_link</parent>
			<child>gps_link</child>
			<pose>0.158613 0 0.079226 0 0 0</pose>	
		</joint>






		<plugin name="thrust_plugin" filename="libthrust_plugin.so">

			<!-- Which thruster curve to use -->
			<mappingType>0</mappingType>

			<!-- General dimensions -->
			<boatWidth>2.4</boatWidth>
			<boatLength>4.9</boatLength>
			<thrustOffsetZ>-0.04</thrustOffsetZ>

			<!-- Thrust specification -->
			<maxCmd>1.0</maxCmd>
			<maxForceFwd>175.0</maxForceFwd>
			<maxForceRev>-150.0</maxForceRev>
			<cmdTimeout>1.0</cmdTimeout>
		</plugin>

		<plugin name="sensors_plugin" filename="libsensors_plugin.so">
		</plugin>

	</model>
</sdf>